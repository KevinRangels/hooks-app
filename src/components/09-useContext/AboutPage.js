import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutPage = () => {

    const { user, setUser } = useContext(UserContext)

    const handleBtn = () => {
        setUser({})
    }


    return (
        <div>
            <h1>About Page</h1>
            <hr/>

            <button
              className="btn btn-primary"
              onClick={handleBtn}
              >
                Logout
            </button>

            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>

        </div>
    )
}
