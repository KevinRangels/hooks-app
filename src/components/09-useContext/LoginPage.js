import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginPage = () => {
  const { user, setUser } = useContext(UserContext);

  return (
    <div>
      <h1>Login Page</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={() =>
          setUser({
            id: 123,
            name: 'Kevin',
          })
        }
      >
        Login
      </button>
    </div>
  );
};
