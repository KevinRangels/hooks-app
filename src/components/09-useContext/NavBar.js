import React from "react";
import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <a className="navbar-brand" href="#">
        Use Context
      </a>
      <button
        className="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarText"
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarText">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item active">
            <NavLink
              exact
              activeClassName="active"
              className="nav-item nav-link"
              to="/"
            >
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-item nav-link"
              to="/about"
            >
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              exact
              activeClassName="active"
              className="nav-item nav-link"
              to="/login"
            >
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
