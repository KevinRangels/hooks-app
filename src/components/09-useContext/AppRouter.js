import React from 'react'
import { NavBar} from './NavBar'
import { AboutPage } from './AboutPage'
import { LoginPage } from './LoginPage'
import { HomePage } from './HomePage'

import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar/>
                <div className="container">
                    <Switch>
                        <Route exact path="/about" component={AboutPage} />
                        <Route exact path="/login" component={LoginPage} />
                        <Route exact path="/" component={HomePage} />
                        <Route component={HomePage} />
                        <Redirect to="/" />
                    </Switch>
                </div>
            </div>
        </Router>
    )
}
