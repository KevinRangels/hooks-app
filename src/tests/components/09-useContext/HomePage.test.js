import { mount } from 'enzyme';
import React from 'react';
import { HomePage } from '../../../components/09-useContext/HomePage';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en componente <HomePage/>', () => {
  const user = {
    name: 'Kevin',
    emial: 'test@mail.com',
  };

  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomePage />
    </UserContext.Provider>,
  );

  test('Debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
