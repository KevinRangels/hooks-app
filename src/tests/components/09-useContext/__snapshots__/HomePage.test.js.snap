// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <HomePage/> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Page
  </h1>
  <hr />
  <pre>
    {
   "name": "Kevin",
   "emial": "test@mail.com"
}
  </pre>
</div>
`;
